src/pages/index.js

```diff
@@ -24,29 +24,151 @@ import {
   previewCloseModal,
   editFormElemenet,
   addFormElemenet,
+  editAvatarForm,
 } from "../utils/constants.js";
-import Card from "../components/card.js";
+import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
 import "../pages/index.css";
 import Section from "../components/Section.js";
 import UserInfo from "../components/Userinfo.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopUpWithImage from "../components/PopupWithImage.js";
+import Api from "../components/Api.js";
+import PopupConfirmDelete from "../components/PopupConfirmDelete.js";
 
+//Api
+
+const api = new Api({
+  authorization: "91a10936-bcf5-444e-a37c-1d53d5865859",
+  "Content-Type": "application/json",
+});
 /* Class Instances */
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
+
+let section; // undefined
+
+api
+  .getUserInfo()
+  .then((data) => {
+    console.log(data); ///take this out this was to make sure right thing was being called
+    userInfo.setUserInfo(data.name, data.about);
+    userInfo.setAvatar(data.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getInitialCards()
+  .then((result) => {
+    section = new Section(
+      {
+        items: result,
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          section.addItem(cardElement);
+          //renderCard(cardData);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+    console.log(result);
+  })
+  .catch((err) => {
+    console.error(err); // log the error to the console
+  });
+
+function handleEditProfileSubmit(inputValues) {
+  /*userInfo.setUserInfo(inputValues.name, inputValues.about);*/
+  profileEditPopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```